src/main/java/uk/gov/justice/hmpps/datacompliance/web/controllers/UalController.java

```diff
@@ -0,0 +1,67 @@
+package uk.gov.justice.hmpps.datacompliance.web.controllers;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.multipart.MultipartFile;
+import uk.gov.justice.hmpps.datacompliance.services.ual.UalService;
+import uk.gov.justice.hmpps.datacompliance.web.dto.ErrorResponse;
+import uk.gov.justice.hmpps.datacompliance.web.dto.UalOffenderResponse;
+
+import java.net.URI;
+import java.util.List;
+
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+import static org.springframework.http.ResponseEntity.created;
+import static org.springframework.http.ResponseEntity.noContent;
+
+@Slf4j
+@RestController
+@AllArgsConstructor
+@Api(tags = {"/ual"})
+@RequestMapping(value = "/ual", produces = APPLICATION_JSON_VALUE)
```

Comment:
It might be worth considering protecting these endpoints with a role otherwise any member of staff could in theory use these endpoints?

Reply:
How are we anticipating this upload to occur? Who is supplying this list?

Reply:
Hey, for context the report will be provided on a quarterly basis by Jackie Cummins. Someone from the GDPR team will then upload it.

I will add some info to the read me with a cURL request to clarify that.